src/cmd-tag

```diff
@@ -0,0 +1,211 @@
+    list_tags.set_defaults(func=cmd_list)
+
+    update = subparsers.add_parser("update", help="Update a tag to new build")
+    update.add_argument("--tag", help="Tag to be updated", required=True)
+    update.add_argument("--build", help="Build to update tag with",
+                        required=True)
+    update.add_argument("--force", help="Force the update of a tag",
+                        action="store_true")
+    update.set_defaults(func=cmd_update)
+
+    return parser.parse_args()
+
+
+def init_build_data():
+    """Initialize build metadata"""
+
+    builds_json = "builds/builds.json"
+
+    if os.path.isfile(builds_json):
+        with open(builds_json) as f:
+            builddata = json.load(f)
+    else:
+        print("INFO: Did not find existing builds.json; starting with empty "
+              "build list and tags list")
+        builddata = {"builds": [], "tags": []}
+
+    return builddata
+
+
+def finalize_json(builddata):
+    """Wrapper around cmdlib.write_json()"""
+
+    builds_json = "builds/builds.json"
+
+    builddata['timestamp'] = rfc3339_time()
+    write_json(builds_json, builddata)
+
+
+def available_builds(builddata):
+    """Returns the availble build IDs from build metadata"""
+
+    return builddata["builds"]
+
+
+def available_tags(builddata):
+    """Returns the available tag names from build metadata"""
+
+    available_tag_names = []
+    if "tags" in builddata.keys():
+        for t in builddata["tags"]:
+            available_tag_names.append(t["name"])
+
+    return available_tag_names
+
+
+def cmd_list(args):
+    """List available tags in build metadata"""
+
+    build_data = init_build_data()
+    avail_tags = available_tags(build_data)
+    if not avail_tags:
+        print("No tags found")
+        sys.exit(0)
+
+    for t in build_data["tags"]:
+        print(f"name: {t['name']}\n"
+              f"created: {t['created']}\n"
+              f"target: {t['target']}\n")
+    sys.exit(0)
+
+
+def cmd_create(args):
+    """Create a new tag from a build ID"""
+
+    # To create a tag, just append to existing list
+    build_data = init_build_data()
+    avail_tags = available_tags(build_data)
+    avail_builds = available_builds(build_data)
+
+    if args.tag in avail_tags:
+        fatal("Cannot create tag that already exists")
+
+    if not args.force and args.build not in avail_builds:
+        fatal("Cannot create a tag with a build that does not exist")
+
+    if args.force:
+        print("INFO: Forcing creation of tag with non-existent "
+              f"build ID: {args.build}")
+
+    if "tags" not in build_data.keys():
+        build_data["tags"] = []
+
+    # Create new tag with timestamp
+    created = rfc3339_time()
+    build_data['tags'].append({"name": args.tag,
+                               "created": created,
+                               "target": args.build})
+
+    # Write out JSON
+    finalize_json(build_data)
+
+
+def cmd_update(args):
```

Comment:
Hmm, wonder if we should just fold `create` into `update`. Otherwise, one has to check first if it exists or not to know if to call `create` or `update`. And then if one somehow doesn't want to overwrite a tag, they can do `list | grep`? Or we could add a `--create-only` flag or something if that's common.

Reply:
Not opposed to this idea.  I'll pursue it.